logrotate.c

```diff
@@ -1210,6 +1215,12 @@ static int copyTruncate(const char *currLog, const char *saveLog, const struct s
     }
 
     if (flags & LOG_FLAG_COPYTRUNCATE) {
+#if defined(WITH_FALLOCATE)
+        if (flags & LOG_FLAG_COPYREDUCE) {
+            message(MESS_ERROR, "copytruncate and copyreduce are mutually exclusive\n");
+            goto fail;
```

Comment:
I propose to make it warning only.  Otherwise logs that use `copytruncate` or `copyreduce` would stop rotating if someone used the other configuration directive on global scope.

Reply:
I see. I removed "goto fail".